test/e2e/resources/version_with_check.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: lambda.services.k8s.aws/v1alpha1
+kind: Version
+metadata:
+  name: $VERSION_NAME
+  annotations:
+    services.k8s.aws/region: $AWS_REGION
+spec:
+  functionName: $FUNCTION_NAME
+  codeSHA256: $HASH
```

Comment:
How is the `$HASH` set

Reply:
`codeSHA256` and `RevisionId` fields are optional and are used to check whether the Version is published only when there is some change in the code or its configuration. I have added new E2E test to verify the code by setting $HASH to publish the Version.